core/src/main/java/org/apache/druid/guice/JsonConfigProvider.java

```diff
@@ -134,6 +153,20 @@
     binder.bind(instanceKey).toProvider(new SupplierProvider<>(supplierKey));
   }
 
+  @SuppressWarnings("unchecked")
```

Comment:
This `unchecked` annotation is useless. FYI, the same issues occurs in line [#143](https://github.com/apache/incubator-druid/pull/6637/files#diff-02fb611063c0b7ecec0bba4ad5f468fbR143). Maybe we should remove them.

Reply:
What do you mean by "useless"? This line is an unchecked cast, and would generate a warning without the annotation:

```
(Key) Key.get(Types.newParameterizedType(Supplier.class, classToProvide))
```

Reply:
Hi, @gianm . Yes, I know what you mean. What I want to say is that this annotation is unnecessary for the other `bind` methods in line 143 and 157 because the casts are not used in these `bind` methods.

line 143:
```java
org.apache.druid.guice.JsonConfigProvider#bind(com.google.inject.Binder, java.lang.String, java.lang.Class<T>, com.google.inject.Key<T>, com.google.inject.Key<com.google.common.base.Supplier<T>>)
```

line 156:
```java
org.apache.druid.guice.JsonConfigProvider#bind(com.google.inject.Binder, java.lang.String, java.lang.Class<T>, java.lang.Class<? extends T>, com.google.inject.Key<T>, com.google.inject.Key<com.google.common.base.Supplier<T>>)
```

Reply:
Ah, ok, you're right, I was looking at the wrong line. Thanks for pointing it out. I fixed it and pushed again.

Reply:
Thanks :+1: